src/main/java/com/gregtechceu/gtceu/api/recipe/chance/logic/ChanceLogic.java

```diff
@@ -162,16 +163,44 @@ public String toString() {
                                                          @NotNull ChanceBoostFunction boostFunction,
                                                          int recipeTier, int chanceTier,
                                                          @Nullable Object2IntMap<?> cache, int times) {
+            // Have to set up a system where all chances are set to be out of 10000
+            List<Integer> chancesOutOfTenThousand = new java.util.ArrayList<>(chancedEntries.stream()
+                    .map(orig -> (int) (((float) orig.chance / (float) orig.maxChance) * (float) getMaxChancedValue()))
+                    .toList());
```

Comment:
use `IntList` instead of a `List<Integer>`. Also, you don't need to explicitly cast all of them, one is enough.
Also, a stream is actually bad to use here because the amount of items in `chancedEntries` is usually small enough that you only get worse performance by using streams.
```suggestion
            IntList chancesOutOfTenThousand = new IntArrayList(chancedEntries.size());
            for (Content orig : chancedEntries) {
                chancesOutOfTenThousand.add((int) ((orig.chance / (float) orig.maxChance) * getMaxChancedValue()));
            }
```